jgrapht-core/src/main/java/org/jgrapht/alg/isomorphism/GraphOrdering.java

```diff
@@ -0,0 +1,269 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* -------------------------
+ * GraphOrdering.java
+ * -------------------------
+ * (C) Copyright 2015, by Fabian Späh and Contributors.
+ *
+ * Original Author:  Fabian Späh
+ * Contributor(s):   Rita Dobler
+ *
+ * $Id$
+ *
+ * Changes
+ * -------
+ * 20-Jun-2015 : Initial revision (FS);
+ *
+ */
+package org.jgrapht.alg.isomorphism;
+
+import org.jgrapht.Graph;
+import org.jgrapht.DirectedGraph;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * This class represents the order on the graph vertices. There are also some
+ * helper-functions for receiving outgoing/incoming edges, etc.
+ * 
+ * @author Fabian Späh
+ * 
+ * @param <V> the type of the vertices
+ * @param <E> the type of the edges
+ */
+
+public class GraphOrdering<V, E>
```

Comment:
Since this class is a helper, it should not be public.  Some aspects of it might be suitable for reuse outside of isomorphism, but we can leave that refactoring work for the future.